### PR TITLE
Wrap sph2grd

### PR DESCRIPTION
**Pull request by willschlitzer**
This pull request wraps the GMT module [`sph2grd`](https://docs.generic-mapping-tools.org/6.2/sph2grd.html).

**Preview** docs at https://pygmt-git-wrap-sph2grd-gmt.vercel.app/api/generated/pygmt.sph2grd.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: doc/api/index.rst [hunk 1]
```diff
@@ -96,6 +96,7 @@ Operations on grids:
     grdgradient
     grdsample
     grdtrack
+    sph2grd
```
Comment (meghanrjones) [inline doc/api/index.rst:99]: This is an operation on tabular data.

File: pygmt/src/sph2grd.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+@fmt_docstring
+@use_alias(
+    G="outgrid",
+    I="increment",
+    R="region",
+    V="verbose",
+)
+@kwargs_to_strings(I="sequence", R="sequence")
+def sph2grd(table, **kwargs):
```
Comment (meghanrjones) [inline pygmt/src/sph2grd.py:23]: Not sure when https://github.com/GenericMappingTools/pygmt/issues/1479 will be decided, but please make sure the input parameter is consistent and include in the docstring.

File: pygmt/src/sph2grd.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                lib.call_module("sph2grd", arg_str)
+
+        if outgrid == tmpfile.name:  # if user did not set outgrid, return DataArray
+            with xr.open_dataarray(outgrid) as dataarray:
+                result = dataarray.load()
+                _ = result.gmt  # load GMTDataArray accessor information
+        else:
+            result = None  # if user sets an outgrid, return None
+
+        return result
```
Comment (meghanrjones) [inline pygmt/src/sph2grd.py:68]:
```suggestion
      return load_dataarray(outgrid) if outgrid == tmpfile.name else None
```

File: pygmt/src/sph2grd.py (hunk 2 above)
Comment (willschlitzer) [reply pygmt/src/sph2grd.py:23]: Updated in 974f908

File: doc/api/index.rst (hunk 1 above)
Comment (willschlitzer) [reply doc/api/index.rst]: Updated in f98323c

File: pygmt/src/sph2grd.py (hunk 2 above)
Comment (weiji14) [reply pygmt/src/sph2grd.py:23]: Time to change it to 'data' as per #1479.
```suggestion
def sph2grd(data, **kwargs):
```

File: pygmt/src/sph2grd.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+@kwargs_to_strings(I="sequence", R="sequence")
+def sph2grd(table, **kwargs):
+    r"""
+    Create spherical grid files in tension of data.
+
+    Reads a spherical harmonics coefficient table with records of L, M,
+    C[L,M], S[L,M] and evaluates the spherical harmonic model on the
+    specified grid.
+
+    {aliases}
```
Comment (weiji14) [inline pygmt/src/sph2grd.py:31]:
```suggestion
    specified grid.

    Full option list at :gmt-docs:`sph2grd.html`

    {aliases}
```

File: pygmt/src/sph2grd.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+    Parameters
+    ----------
+    table : str or {table-like}
```
Comment (weiji14) [inline pygmt/src/sph2grd.py:35]:
```suggestion
    data : str or {table-like}
```

File: pygmt/src/sph2grd.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        in.
+    {I}
+    {R}
+    {V}
+
+    Returns
+    -------
+    ret: xarray.DataArray or None
+        Return type depends on whether the ``outgrid`` parameter is set:
+        - :class:`xarray.DataArray` if ``outgrid`` is not set
```
Comment (weiji14) [inline pygmt/src/sph2grd.py:50]:
```suggestion
        Return type depends on whether the ``outgrid`` parameter is set:

        - :class:`xarray.DataArray` if ``outgrid`` is not set
```

File: pygmt/src/sph2grd.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        - None if ``outgrid`` is set (grid output will be stored in file set by
+          ``outgrid``)
+    """
+    with GMTTempFile(suffix=".nc") as tmpfile:
+        with Session() as lib:
+            file_context = lib.virtualfile_from_data(check_kind="vector", data=table)
```
Comment (weiji14) [inline pygmt/src/sph2grd.py:56]:
```suggestion
            file_context = lib.virtualfile_from_data(check_kind="vector", data=data)
```

File: pygmt/tests/test_sph2grd.py [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+from pygmt.helpers import GMTTempFile
+
+
+def test_sph2grd_outgrid():
+    """
+    Test sph2grd with a set outgrid.
+    """
+    with GMTTempFile(suffix=".nc") as tmpfile:
+        result = sph2grd(
+            table="@EGM96_to_36.txt", outgrid=tmpfile.name, increment=1, region="g"
```
Comment (weiji14) [inline pygmt/tests/test_sph2grd.py:17]: Remember to add `@EGM96_to_36.txt` to the cache at https://github.com/GenericMappingTools/pygmt/blob/8eb055d2a696c59da2cbbe57b473885562aeb3a9/pygmt/helpers/testing.py#L167
```suggestion
            data="@EGM96_to_36.txt", outgrid=tmpfile.name, spacing=1, region="g"
```

File: pygmt/tests/test_sph2grd.py [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+        )
+        assert result is None  # return value is None
+        assert os.path.exists(path=tmpfile.name)  # check that outgrid exists
+
+
+def test_sph2grd_no_outgrid():
+    """
+    Test sph2grd with no set outgrid.
+    """
+    temp_grid = sph2grd(table="@EGM96_to_36.txt", increment=1, region="g")
```
Comment (weiji14) [inline pygmt/tests/test_sph2grd.py:27]:
```suggestion
    temp_grid = sph2grd(data="@EGM96_to_36.txt", spacing=1, region="g")
```

File: pygmt/src/sph2grd.py [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+
+
+@fmt_docstring
+@use_alias(
+    G="outgrid",
+    I="increment",
```
Comment (weiji14) [inline pygmt/src/sph2grd.py:18]:
```suggestion
    I="spacing",
```

File: pygmt/src/sph2grd.py [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+)
+from pygmt.io import load_dataarray
+
+
+@fmt_docstring
+@use_alias(
+    G="outgrid",
+    I="spacing",
+    R="region",
+    V="verbose",
```
Comment (weiji14) [inline pygmt/src/sph2grd.py:21]: Add common aliases.
```suggestion
    V="verbose",
    b="binary",
    h="header",
    i="incols",
    r="registration",
    x="cores",
```

File: pygmt/src/sph2grd.py [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+)
+@kwargs_to_strings(I="sequence", R="sequence")
```
Comment (weiji14) [inline pygmt/src/sph2grd.py:23]:
```suggestion
@kwargs_to_strings(I="sequence", R="sequence", i="sequence_comma")
```

File: pygmt/src/sph2grd.py [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+    data : str or {table-like}
+        Pass in (x, y, z) or (longitude, latitude, elevation) values by
+        providing a file name to an ASCII data table, a 2D
+        {table-classes}.
+    outgrid : str or None
+        The name of the output netCDF file with extension .nc to store the grid
+        in.
+    {I}
+    {R}
+    {V}
```
Comment (weiji14) [inline pygmt/src/sph2grd.py:47]:
```suggestion
    {V}
    {b}
    {h}
    {i}
    {r}
    {x}
```

**Review by weiji14 — APPROVED** [review]
Looks ready for final review. Just some minor comments to fix the lint error and add some common aliases.

Also, remember to cache the `@EGM96_to_36.txt` dataset as mentioned in https://github.com/GenericMappingTools/pygmt/pull/1434#discussion_r715994606. You can do it directly in this PR instead of opening a new one.

File: pygmt/src/sph2grd.py [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+    Full option list at :gmt-docs:`sph2grd.html`
+
+    {aliases}
+
+    Parameters
+    ----------
+    data : str or {table-like}
+        Pass in (x, y, z) or (longitude, latitude, elevation) values by
+        providing a file name to an ASCII data table, a 2D
+        {table-classes}.
```
Comment (seisman) [inline pygmt/src/sph2grd.py:45]: I have never used this module, but I believe it doesn't need (x, y, z) input.

File: pygmt/src/sph2grd.py [hunk 15]
```diff
@@ -0,0 +1,78 @@
...
+    """
+    with GMTTempFile(suffix=".nc") as tmpfile:
+        with Session() as lib:
+            file_context = lib.virtualfile_from_data(check_kind="vector", data=data)
+            with file_context as infile:
+                if "G" not in kwargs.keys():  # if outgrid is unset, output to tempfile
+                    kwargs.update({"G": tmpfile.name})
+                outgrid = kwargs["G"]
+                arg_str = build_arg_string(kwargs)
+                arg_str = " ".join([infile, arg_str])
```
Comment (seisman) [inline pygmt/src/sph2grd.py:75]:
```suggestion
                arg_str = " ".join([infile, build_arg_string(kwargs)])
```

File: pygmt/src/sph2grd.py (hunk 14 above)
Comment (willschlitzer) [reply pygmt/src/sph2grd.py]: updated in 4f22a4e

**Review by seisman — APPROVED** [review]

**Pull request merged**